### ISSUE TITLE
please make line chart drawDots optional

### ISSUE THREAD

**Issue by davisford** [dc-js/dc.js#689]
If you turn the brush off on a line chart, it will create an extra group with dots and when you mouseover the dot it will show itself and show x/y lines back to the axes. 

I want to do something similar, but not quite the same, and now I must either fork and turn this off, or else, remove it post render.  

I think this behavior would be better served if it were configurable -- just like `brushOn( )`, add `dotsOn( )` function to opt-in to this behavior.

**Comment by mtraynham** [comment]
I was working with this a bit @gordonwoodhull. I think that PR doesn't satisfy all the cases as the Brush for line charts is drawn on the DOM after the chart body and therefore intercepts the pointer events on the dots.

**Comment by gordonwoodhull** [comment]
Right, #735 only controls whether the dots will show when the brush is off. You are looking for the ability to turn the dots on when the brush is also on?

**Comment by mtraynham** [comment]
Yeah, I locally changed some of the CSS to turn off pointer events on the lines/areas that get generated, but the brush is still on top of the dots.  The weirder part is that a composite chart, generates the child `g` on redraw, so after someone filters a different chart, the stacks come to the top and the dots **are** interactive with brushing.

**Comment by gordonwoodhull** [comment]
No doubt, that's a more difficult problem, which is likely why the design has been either/or up until now.

**Comment by gordonwoodhull** [comment]
It might also be confusing to have little dots interfering with the brush action? I'm not sure.

**Comment by mtraynham** [comment]
The brush remains in drag mode when you hover on the dots, so it's not terri-bad.  I don't think there's a right answer here, so I'm not going to create a PR.  In the mean time, I can work around it in this case with a `.on('postRender',...`, select the composite sub charts and move them in front of the brush, but also disabling pointer events on the lines/areas.

**Comment by gordonwoodhull** [comment]
How did you turn the dots on, though? I thought #735 only provided a way to turn them off when the brush is off.

**Comment by mtraynham** [comment]
Ha, ok, good question.  Still using composite chart here...  the child charts all have brushes, so in the composite chart's `chart` function, I return a new line chart with brushing always set to false.

```
            child.brushOn = function () {
                if (!arguments.length) {
                    return false;
                }
                return this;
            };
```

**Comment by gordonwoodhull** [comment]
Wow. The leaky abstraction that is dc.js.

**Comment by mtraynham** [comment]
You saw nothing! :smile:

**Comment by gordonwoodhull** [comment]
This was implemented in #735; continuing discussion of the opposite feature in #1152.
